### ISSUE TITLE
Updating a group permission, updates wrong group permission

### ISSUE THREAD

**Issue by ricricucit** [directus/directus#1358]
## Version Info
  - Directus version and branch: development
  - PHP version: 7.0.15
  - MySQL version: 5.6.35
  - Web server: Apache
  - OS name and version: OSX

## Expected Behavior
Changing permission to a specific group, should change the permissions to that particular group.

## Actual Behavior
Changing permission to a specific group, changes the permissions to the group that is modifying the permission (in this case, admin). See screenshots and step to reproduce its.

## Steps to Reproduce
 1. Create a Group (eg. "Editor", like in the screenshot)
 2. Using an Administrator user hide/show, for example "Directus Preferences" for the "Editor Group"
 3. You'll see that even tho you are in the Administrator, the menu on the left side hides/shows the "Directus Preferences" table, which should be hiding/showing for another users' group (Editor)

## Logs or Screenshots
In the screenshots a member of the Administrator group is modifying the permissions for the Editor group for "Directus Preferences"...but on the left side, you can see that the Directus Preferences disappear/appear for himself (Administrator)...and not for the target group (Editor).

<img width="779" alt="screen shot 2017-02-25 at 20 13 04" src="https://cloud.githubusercontent.com/assets/1016775/23334063/82248256-fb97-11e6-9fbe-eeae0786c801.png">
<img width="779" alt="screen shot 2017-02-25 at 20 13 18" src="https://cloud.githubusercontent.com/assets/1016775/23334064/82269168-fb97-11e6-83a7-0c40aec1ef49.png">

If I keep "playing around" with this options, DB data gets corrupted and see stuff like this:

<img width="723" alt="screen shot 2017-02-25 at 20 22 55" src="https://cloud.githubusercontent.com/assets/1016775/23334098/4a922220-fb98-11e6-9af8-02c511c474dc.png">

**UPDATE:**
I just noticed that it's simply a weird behaviour (maybe a feature?) that –temporarily– shows the tables items with "view all" capabilities in the menu. When refreshed, they disappear (which I'm guessing it's the correct way of representing the UI).

Q: Playing with this I noticed the `nav_listed` field in the DB; is that actually used to list or not list things like tables, bookmarks and system menus' items?
I'm asking because, trying, I could not hide the "Messages" menu item by simply setting that to 0 (for the right users_group).

**Comment by ricricucit** [comment]
Now I'm getting the following, if I login with the "Editor" user (and can't fix this issue anymore, even if I try to manually change data directly in the DB):

````
{
  "code":0,
  "class":"TypeError",
  "message":"Argument 2 passed to Directus\\Db\\TableGateway\\BaseTableGateway::convertDates() must be of the type array, boolean given, called in \/Applications\/MAMP\/htdocs\/welance-pfolio\/api\/core\/Directus\/Db\/TableGateway\/BaseTableGateway.php on line 562",
  "file":"\/Applications\/MAMP\/htdocs\/welance-pfolio\/api\/core\/Directus\/Db\/TableGateway\/BaseTableGateway.php",
  "line":517,
  "trace":[
    {
      "file":"\/Applications\/MAMP\/htdocs\/welance-pfolio\/api\/core\/Directus\/Db\/TableGateway\/BaseTableGateway.php",
      "line":562,
      "function":"convertDates",
      "class":"Directus\\Db\\TableGateway\\BaseTableGateway",
      "type":"->",
      "args":[
        [
          
        ],
        false,
        "directus_privileges"
      ]
    },
    {
      "file":"\/Applications\/MAMP\/htdocs\/welance-pfolio\/api\/core\/Directus\/Db\/TableGateway\/DirectusPrivilegesTableGateway.php",
      "line":221,
      "function":"parseRecord",
      "class":"Directus\\Db\\TableGateway\\BaseTableGateway",
      "type":"->",
      "args":[
        [
          
        ]
      ]
    },
    {
      "file":"\/Applications\/MAMP\/htdocs\/welance-pfolio\/index.php",
      "line":339,
      "function":"fetchGroupPrivilegesRaw",
      "class":"Directus\\Db\\TableGateway\\DirectusPrivilegesTableGateway",
      "type":"->",
      "args":[
        null
      ]
    },
    {
      "file":"\/Applications\/MAMP\/htdocs\/welance-pfolio\/index.php",
      "line":453,
      "function":"getPrivileges",
      "args":[
        null
      ]
    }
  ],
  "traceAsString":"#0 \/Applications\/MAMP\/htdocs\/welance-pfolio\/api\/core\/Directus\/Db\/TableGateway\/BaseTableGateway.php(562): Directus\\Db\\TableGateway\\BaseTableGateway->convertDates(Array, false, 'directus_privil...')\n#1 \/Applications\/MAMP\/htdocs\/welance-pfolio\/api\/core\/Directus\/Db\/TableGateway\/DirectusPrivilegesTableGateway.php(221): Directus\\Db\\TableGateway\\BaseTableGateway->parseRecord(Array)\n#2 \/Applications\/MAMP\/htdocs\/welance-pfolio\/index.php(339): Directus\\Db\\TableGateway\\DirectusPrivilegesTableGateway->fetchGroupPrivilegesRaw(NULL)\n#3 \/Applications\/MAMP\/htdocs\/welance-pfolio\/index.php(453): getPrivileges(NULL)\n#4 {main}"
}
````

**Comment by benhaynes** [comment]
Het @ricricucit – part of this is a bug, yes. To answer your question, there is a `nav_listed` toggle that allows you to keep `view` enabled (eg: inside relational interfaces) but removes the table from the left nav. You can also use the group's nav_blacklist to accomplish something similar.

@WellingGuzman – When adjusting group permissions, it should only immediately update the GUI (eg: hide the table in the nav) if you are editing YOUR group (always the admin group as of now, since only they can edit permissions).

The error is that if I (admin) update another group (managers) table to VIEW = FALSE ... it updates the nav to hide it for ME too... which it shouldn't since I'm not updating my group.

**Comment by benhaynes** [comment]
@ricricucit – thanks for the update. We're talking about the `nav_listed` versus permissions over on this ticket too: https://github.com/directus/directus/issues/1625

I would like to have the permissions control everything, but in some instances it's good to have a global option (so you don't have to change something for every group). I'd love to get your thoughts on all this!

**Comment by WellingGuzman** [comment]
Is this issue marked `wontfix` for 6.3? @ricricucit is this still happening to you in 6.4?

**Comment by benhaynes** [comment]
@WellingGuzman Can you give this an *hour* of research and testing to see if it's an issue or resolved? If we can't replicate, let's close since it's been a few months with no response.

**Comment by WellingGuzman** [comment]
Okay, I checked on 6.4, it actually toggle the visibility of any table on any group to the admin group (which is the only group allowed to edit permissions).

I removed the syncing part, which will avoid this problem.  It didn't actually change the permission on the group's privilege on the database.

It won't be fixed on 6.3.

Closed by https://github.com/directus/directus/commit/696bbdfcee4a56ff4bd98e8ce34422d85b7c8e74
